content/en/guides/core/artifacts/track-external-files.md

```diff
@@ -9,82 +8,82 @@ title: Track external files
 s3://my-bucket
-+-- datasets/
-|		+-- mnist/
-+-- models/
-		+-- cnn/
+
+|datasets/
+  |---- mnist/
+|models/
+  |---- cnn/
 ```
 
-Under `mnist/` we have our dataset, a collection of images. Lets track it with an artifact:
+Within `mnist/` is a collection of images. Track the your images dataset with `wandb.Artifact.add_reference()`:
 
 ```python
 import wandb
 
 run = wandb.init()
-artifact = wandb.Artifact("mnist", type="dataset")
-artifact.add_reference("s3://my-bucket/datasets/mnist")
+artifact = wandb.Artifact(name="mnist", type="dataset")
+artifact.add_reference(uri="s3://my-bucket/datasets/mnist")
 run.log_artifact(artifact)
+run.finish()
 ```
-{{% alert color="secondary" %}}
-By default, W&B imposes a 10,000 object limit when adding an object prefix. You can adjust this limit by specifying `max_objects=` in calls to `add_reference`.
-{{% /alert %}}
-
-Our new reference artifact `mnist:latest` looks and behaves similarly to a regular artifact. The only difference is that the artifact only consists of metadata about the S3/GCS/Azure object such as its ETag, size, and version ID (if object versioning is enabled on the bucket).
 
-W&B will use the default mechanism to look for credentials based on the cloud provider you use. Read the documentation from your cloud provider to learn more about the credentials used:
+The previous code sample creates a reference artifact `mnist:latest`. Within the W&B App, you can look through the contents of the reference artifact using the file browser, [explore the full dependency graph]({{< relref "/guides/core/artifacts/explore-and-traverse-an-artifact-graph" >}}), and scan through the versioned history of your artifact. The W&B App does not render rich media such as images, audio, and so forth because the data itself is not contained within the artifact.
 
-| Cloud provider | Credentials Documentation |
-| -------------- | ------------------------- |
-| AWS            | [Boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials) |
-| GCP            | [Google Cloud documentation](https://cloud.google.com/docs/authentication/provide-credentials-adc) |
-| Azure          | [Azure documentation](https://learn.microsoft.com/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) |
-
-For AWS, if the bucket is not located in the configured user's default region, you must set the `AWS_REGION` environment variable to match the bucket region.
-
-Interact with this artifact similarly to a normal artifact. In the App UI, you can look through the contents of the reference artifact using the file browser, explore the full dependency graph, and scan through the versioned history of your artifact.
+{{% alert %}}
+W&B Artifacts support any Amazon S3 compatible interface, including MinIO. The scripts described below work as-is with MinIO, when you set the `AWS_S3_ENDPOINT_URL` environment variable to point at your MinIO server.
```

Comment:
You could steal the language and link for MiniIO from https://docs.wandb.ai/guides/hosting/data-security/secure-storage-connector/#availability-matrix.